contracts/Treasury.sol

```diff
@@ -181,6 +183,8 @@ contract Treasury is Initializable, AccessControlUpgradeable {
         IERC20(partneredProjects[partnerId].tokenAddress).safeTransfer(msg.sender, getAmountWithAdjustedDecimals(partnerTokenAmount, partnerTokenDecimals));
 
         multiplierTimestamp[partnerId] = multiplierTimestamp[partnerId].add(partnerTokenAmount.div(partneredProjects[partnerId].tokenSupply.div(projectDistributionTime[partnerId])).div(uint(1e18).div(multiplierUnit)));
+
+        emit TreasuryClaimed(msg.sender, partnerId, partnerTokenAmount, currentMultiplier);
```

Comment:
Current multiplier is a reference value to check slippage and isn't always the real multiplier. Effective and most precise multiplier value is `getProjectMultiplier(partnerId)` (also used in slippage check in line 167).

This function logic affects the multiplier value (returned by `getProjectMultiplier(partnerId)`) so you would need to store it before it's changed.

I'd say store the value before slippage check like
`uin256 effectiveMultiplier = getProjectMultiplier(partnerId);`
Then use it in both slippage check and event.